python/cudf/cudf/core/column/column.py

```diff
@@ -2719,8 +2731,15 @@ def as_column(
             return as_column(arbitrary, dtype=dtype, nan_as_null=nan_as_null)
         elif arbitrary.dtype.kind in "biuf":
             from_pandas = nan_as_null is None or nan_as_null
-            return as_column(
+            if not is_np_native_byteorder(arbitrary.dtype):
+                # Not supported by pyarrow
+                arbitrary = arbitrary.astype(arbitrary.dtype.newbyteorder())
```

Comment:
If it's not native byteorder then swapping byteorder will get you native byteorder - I'd find specifying native byteorder explicitly would be a bit more explicit though:

```suggestion
                arbitrary = arbitrary.astype(arbitrary.dtype.newbyteorder("="))
```